src/main/java/com/mastery/java/task/controllers/EmployeeRestController.java

```diff
@@ -0,0 +1,114 @@
+        log.trace("Getting all employees from the database");
+        logCon.trace("Getting all employees from the database");
+        return ResponseEntity.ok(employees);
+    }
+
+    @GetMapping("/getById")
+    @ApiOperation("method to get employee bu id")
+    public ResponseEntity<Employee> getEmployeeById(@RequestParam("id") Integer id) {
+        Employee employee = employeeService.getById(id);
+        log.info("Employee with ID " + id + " received: " + employee);
+        logCon.info("Employee with ID " + id + " received: " + employee);
+        return ResponseEntity.ok(employee);
+    }
+
+    @GetMapping("/add")
+    @ApiOperation("method to add employee")
+    public ResponseEntity<?> addEmployee(@Valid Employee employee, BindingResult bindingResult) throws DuplicateEmployeeException, InvalidDateException {
+        System.out.println(employee);
+        if (bindingResult.hasErrors()) {
+            log.info("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            log.trace("Invalid employee: " + bindingResult.getAllErrors());
+            logCon.info("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            return ResponseEntity.ok(collectResponseToInvalidRequest(employee, bindingResult));
+        }else {
+            employeeService.save(employee);
+            return ResponseEntity.ok(employee);
+        }
+    }
+
+    @GetMapping("/update")
+    @ApiOperation("method to update employee")
+    public ResponseEntity<?> updateEmployee(@RequestParam("id") Integer id, @Valid Employee employee, BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            log.warn("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            log.debug("Invalid employee: " + bindingResult.getAllErrors());
+            logCon.warn("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            return ResponseEntity.ok(collectResponseToInvalidRequest(employee, bindingResult));
+        }
+        employee.setEmployeeId(id);
+        employeeService.update(employee);
+        log.info("Employee with ID " + id + " changed: " + employee);
+        logCon.info("Employee with ID " + id + " changed: " + employee);
+        return ResponseEntity.ok(employee);
+    }
+
+    @GetMapping("/delete")
+    @ApiOperation("method to delete employee by id")
+    public ResponseEntity<Employee> deleteEmployee(@RequestParam("id") Integer id) {
+        Employee employee = employeeService.getById(id);
+        this.jmsMessagingTemplate.convertAndSend(this.queue, String.valueOf(id));
+        log.info("Employee with ID " + id + " deleted");
+        logCon.info("Employee with ID " + id + " deleted");
+        return ResponseEntity.ok(employee);
+    }
+
+    public String collectResponseToInvalidRequest(Employee employee, BindingResult bindingResult) {
+        String str = new String("Запрос имеет некорректные данные в след. полях :\n");
```

Comment:
please use more meaningful names for variables. please use english language for notifications

Reply:
fixed it